todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/domain/event.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.android.architecture.blueprints.todoapp.taskdetail.domain
+
+/** Things we react to **/
+sealed class TaskDetailEvent {
+    object DeleteTaskRequested : TaskDetailEvent()
+    object CompleteTaskRequested : TaskDetailEvent()
+    object ActivateTaskRequested : TaskDetailEvent()
+    object EditTaskRequested : TaskDetailEvent()
+    object TaskDeleted : TaskDetailEvent()
+    object TaskMarkedComplete : TaskDetailEvent()
+    object TaskMarkedActive : TaskDetailEvent()
+    object TaskSaveFailed : TaskDetailEvent()
+    object TaskDeletionFailed : TaskDetailEvent()
+
+    companion object {
+        @JvmStatic fun deleteTaskRequested() = DeleteTaskRequested
```

Comment:
Change this to
```
 @JvmStatic fun deleteTaskRequested() = DeleteTaskRequested as TaskDetailEvent
```
So you don't have to do it in the tests since those still haven't been converted to Java. The methods generated by DataEnum return the super type not the subtype so this is actually breaking the expected Java API